### PR TITLE
fix : Update Event dates on timezone preference change

### LINKED ISSUE #1892: Event dates is not being updated on Timezone preference change
**Actual Behaviour**

Currently when user changes timezone preference from settings activity. It is not updating event dates.
It is possible that for one time zone date is `9` and for other time zone it is `10`

example.
For 
startsAt = "2017-11-09T23:08:56-03:30";
endsAt = "2017-11-12T09:23:45-03:30";

In Event Time zone(-03:30) Dates are:
[EventDates(date=2017-11-09), EventDates(date=2017-11-10), EventDates(date=2017-11-11), EventDates(date=2017-11-12)]

In Local TimeZone (in my case +05:30(Asia/kolkata)) Dates are:
[EventDates(date=2017-11-10), EventDates(date=2017-11-11), EventDates(date=2017-11-12)]

**Expected Behaviour**

Event dates should be updated on timezone preference change.

**Would you like to work on the issue?**

Yes

**Target Milestone to solve the issue**

[Milestone 16](https://github.com/fossasia/open-event-android/milestone/39)

### PR DESCRIPTION
**Pull request by Shailesh351**
Fixes #1892 

Changes:
- Update event dates in the database on changing timezone preference in settings
- Changed ParseException to DateTimeParseException because ZonedDateTime.parse throws DateTimeParseException

### CODE REVIEW

**Comment by Shailesh351** [comment]
@iamareebjamal @heysadboy  Please review

File: android/app/src/main/java/org/fossasia/openevent/activities/SettingsActivity.java [hunk 1]
```diff
@@ -169,6 +176,22 @@ private void setToolbar() {
             getSupportActionBar().setElevation(Utils.dpToPx(4));
```
Comment (scandindian) [inline android/app/src/main/java/org/fossasia/openevent/activities/SettingsActivity.java:176]: Why setting the elevation dynamically? I think we could use AppBarLayout. It's fine anyways.  

-- android.support.design.widget.AppBarLayout

Then toolbar inside it.

Comment (iamareebjamal) [reply android/app/src/main/java/org/fossasia/openevent/activities/SettingsActivity.java:176]: @heysadboy This change is not part of this PR

File: android/app/src/main/java/org/fossasia/openevent/activities/SettingsActivity.java [hunk 2]
```diff
@@ -169,6 +176,22 @@ private void setToolbar() {
...
+
+        Observable.fromCallable(() ->
+                DateConverter.getDaysInBetween(startTime, endTime)
+        ).subscribe(eventDates -> RealmDataRepository.getDefaultInstance().saveEventDates(eventDates).subscribe(), throwable -> {
+            Timber.e(throwable);
+            Timber.e("Error updating event dates : enable to parse start date: %s and end date: %s in ISO format",
+                    startTime, endTime);
+            OpenEventApp.postEventOnUIThread(new RetrofitError(new Throwable("Error parsing dates")));
+        });
+    }
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/SettingsActivity.java]: This whole block is synchronous, even the fromCallable method. So, there is no need for the fromCallable method.

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**